store/driver/txn/unionstore_driver.go

```diff
@@ -75,7 +77,8 @@ func newUnionstoreBatchBufferGetter(m kv.MemBuffer) unionstore.BatchBufferGetter
 }
 
 func (g *tikvMemBuffer) Get(k []byte) ([]byte, error) {
-	return g.MemBuffer.Get(context.Background(), k)
+	data, err := g.MemBuffer.Get(context.Background(), k)
+	return data, toTiDBErr(err)
```

Comment:
I am a little bit worried that calling `toTiDBErr` would cause some performance regression if these methods are called frequently.

Theoretically, the extra code can be optimized out after auto-inlining. But I think Go's compiler is too naive to do the optimizations.

Reply:
And it seems calling `toTiDBErr` here is duplicated with the calling in `kvGetter`'s methods.